server/manager/src/main/java/org/apache/accumulo/manager/tableOps/Utils.java

```diff
@@ -166,8 +168,17 @@ private static Lock getLock(ServerContext context, AbstractId<?> id, long tid,
     var fLockPath =
         FateLock.path(context.getZooKeeperRoot() + Constants.ZTABLE_LOCKS + "/" + id.canonical());
     FateLock qlock = new FateLock(context.getZooReaderWriter(), fLockPath);
-    Lock lock = DistributedReadWriteLock.recoverLock(qlock, lockData);
-    if (lock == null) {
+    DistributedLock lock = DistributedReadWriteLock.recoverLock(qlock, lockData);
+    if (lock != null) {
+
+      // Validate the recovered lock type
+      boolean isWriteLock = lock.getType() == LockType.WRITE;
+      if (writeLock != isWriteLock) {
+        throw new IllegalStateException("Unexpected lock type " + lock.getType()
+            + " recovered for transaction " + tid + " on object " + id + ". Expected "
```

Comment:
Since we don't have a fate id type, its good to use the following method to format fate ids.  This makes it possible to easily find all messages related to the same fate operation.  Otherwise some could be in decimal and others in hex.  May need to import FateTxId, so this change may not compile.

```suggestion
            + " recovered for transaction " + FateTxId.formatTid(tid) + " on object " + id + ". Expected "
```

Reply:
@rsingh433 please make this change to the branch (including any necessary imports)